### ISSUE TITLE
Rogue Kick @ focus target switching main target

### ISSUE THREAD

**Issue by lingxDEV** [TrinityCore/TrinityCore#24629]
**Description:**

Using focus macro kick switches target.

**Current behaviour:**

When using Kick as a rogue on focus target, your main target is switched to the focus target.

**Expected behaviour:**

When using Kick on focus target, your main target should stay the same and not switch to the focus target.

**Steps to reproduce the problem:**

1. Create a rogue
2. Make the macro /cast [target=focus] Kick
3. Set a main target and a focus target
4. Use the macro
5. Focus target becomes main target

**Branch(es):**

3.3.5

**TC rev. hash/commit:** 

07842547b4f4

**Operating system:**

Windows 10 64

**Comment by jackpoz** [comment]
you can get the hash/commit using ".server debug" chat command (you can run it in the console too)

**Comment by lingxDEV** [comment]
Done and edited in!

**Comment by Jildor** [comment]
I can't reproduce it

**Comment by lingxDEV** [comment]
Ah I see you have to be auto attacking, and it won't change the targets. And I don't see a situation where you wouldn't be.

Makes sense. I'll close this issue.

**Comment by Shauren** [comment]
No, your existing target should never be changed by the server, even if you are not autoattacking (this should only happen if you currently have no target)

**Comment by lingxDEV** [comment]
Then I should probably add, that it happens with a couple of other spells too:

- Cheap Shot
- Garrote
- Kidney Shot (when would I ever Kidney Shot focus? during dance with Premeditation)
- Shiv
- Expose Armor
- Throw

**Comment by Jildor** [comment]
if shauren is right, seems a generic problem related with melee spells.
server-side shouldn't force switch target if you have selected one.

**Comment by CraftedRO** [comment]
1085643b44525df8215b464198751b6f2f174946 still valid, tested on mop classic ptr 5.5.2.63832:

https://github.com/user-attachments/assets/0f742d8a-ab40-45c7-b767-c1c9ef12f190

**Comment by lingxDEV** [comment]
I'm not sure we are on the same page, because the video does not showcase the problem described in my first post.

The problem I'm describing would be if you moved closer to your focused target in the video and actually fired the macro, it should change your current target to the focused target (if the bug exists).
